### PR TITLE
Release 1.4.0 - Support of ink! 3.0.0-rc10

### PR DESCRIPTION
**Pull request by yarikbratashchuk**
After fixing of:
https://github.com/paritytech/ink/pull/1131
https://github.com/paritytech/ink/issues/982
https://github.com/paritytech/ink/pull/1145

We are able to use the latest ink! =) At the moment it points to `master`, but then the `3.0.0-rc10` release will be ready we will point on it.

Updated contract to be compatible with latest `contract-pallet`.
Simplified the brush's codegen, now we don't need to worry about the `ink-as-dependency` feature.

Of course, reduction of the size of the code!=) All contracts are moved to use `Mapping` instead of  `StorageHashMap`. All values are passed by reference instead of move.

At the moment only `substrate-contracts-node` works correctly with the latest ink, so integration tests take a lot of time.

Most of the documentation is updated, but not all. We will update it fully in follow-up PR.

### CODE REVIEW

**Comment by xgreenx** [comment]
You also will be useful to check that PR: https://github.com/Supercolony-net/openbrush-contracts/pull/49/files

**Comment by yarikbratashchuk** [comment]
Okay, I'm leaving it as it is.
My contribution to this was insignificant anyway. 
Now what is needed is to build a working Europa node and make redspot tests pass.

File: examples/psp1155/lib.rs [hunk 1]
```diff
@@ -8,14 +8,14 @@ pub mod my_psp1155 {
...
+    use ink_storage::Mapping;
 
     #[derive(Default, PSP1155Storage)]
     #[ink(storage)]
     pub struct MyPSP1155 {
         #[PSP1155StorageField]
         psp1155: PSP1155Data,
-        denied_ids: StorageHashMap<Id, ()>,
+        denied_ids: Mapping<Id, ()>,
     }
```
Comment (VargSupercolony) [inline examples/psp1155/lib.rs:19]: Maybe we could use some lightweight collection here? As we only need keys some `Set` would be ideal

**Review by VargSupercolony — CHANGES_REQUESTED** [review]
great work! left a couple of comments.

**Review by coreggon11 — APPROVED** [review]

**Review by PierreOssun — APPROVED** [review]
Great work!

File: Cargo.toml [hunk 2]
```diff
@@ -13,7 +13,7 @@ exclude = [
 
 [package]
 name = "brush"
-version = "1.3.0"
+version = "1.4.0"
 authors = ["Supercolony <green.baneling@supercolony.net>"]
 edition = "2018"
```
[the remark this answers is not in the document]
Comment (xgreenx) [reply Cargo.toml:18]: It is `2018` because it doesn't work correctly with tests and `--all-features` with `2021`.

**Pull request merged**